Tests/WhisperKitTests/TestUtils.swift

```diff
@@ -234,3 +234,41 @@ extension SpecialTokens {
         )
     }
 }
+
+extension Result {
+    var isSuccess: Bool {
+        switch self {
+        case .success:
+            return true
+        case .failure:
+            return false
+        }
+    }
+
+    func whisperError() -> WhisperError? {
+        switch self {
+        case .success:
+            return nil
+        case .failure(let error):
+            return error as? WhisperError
+        }
+    }
+}
+
+extension Result where Success == [TranscriptionResult] {
+    func normalizedText(prefix: Int) throws -> String {
+        try get().text.normalized.split(separator: " ").prefix(prefix).joined(separator: " ")
+    }
+}
+
+extension Collection where Element == TranscriptionResult {
+    var text: String {
+        map(\.text).joined(separator: " ")
+    }
+}
+
+extension Collection where Element == TranscriptionResult {
+    var segments: [TranscriptionSegment] {
+        flatMap(\.segments)
+    }
+}
```

Comment:
Aren't some of these needed for the result handling in core library? If so, perhaps these should be in the extension section as well

Reply:
it's needed just for tests so I'd keep it here for now